lib/coinbase/transfer.rb

```diff
@@ -166,7 +168,15 @@ def wait!(interval_seconds = 0.2, timeout_seconds = 10)
       start_time = Time.now
 
       loop do
-        return self if status == Status::COMPLETE || status == Status::FAILED
+        update = Coinbase.call_api do
+          transfers_api.get_transfer(wallet_id, from_address_id, id)
+        end
+
+        if update.status.to_s == Status::COMPLETE.to_s || update.status == Status::FAILED.to_s
+          self.transaction_hash = update.transaction_hash
+          self.status = update.status
+          return self
```

Comment:
Can we just update the `model` and then have `transaction_hash` and `status` be derived from that?

Reply:
this is how the latest commit handles the update